npm/npm-debug-cli/dataplaneParser/parser/parser.go

```diff
@@ -0,0 +1,338 @@
+package parser
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/Azure/azure-container-networking/npm/npm-debug-cli/dataplaneParser/iptable"
+	"github.com/Azure/azure-container-networking/npm/util"
+)
+
+type Parser struct {
+}
+
+// CreateIptableObject create a Go object from specified iptable. Optional read from iptable-save file
+func (p *Parser) ParseIptablesObject(tableName string, filenames ...string) *iptable.Iptables {
+	iptableBuffer := bytes.NewBuffer(nil)
+	if len(filenames) > 0 {
+		byteArray, err := ioutil.ReadFile(filenames[0])
+
+		if err != nil {
+			fmt.Print(err)
+		}
+		for _, b := range byteArray {
+			iptableBuffer.WriteByte(b)
+		}
+		chains := p.parseIptablesChainObject(tableName, iptableBuffer.Bytes())
+		return iptable.NewIptables(tableName, chains)
+
+	} else {
+		cmdArgs := []string{"-t", string(tableName)}
+		cmd := exec.Command(util.IptablesSave, cmdArgs...)
+
+		cmd.Stdout = iptableBuffer
+		stderrBuffer := bytes.NewBuffer(nil)
+		cmd.Stderr = stderrBuffer
+
+		err := cmd.Run()
+
+		if err != nil {
+			_, err = stderrBuffer.WriteTo(iptableBuffer)
+			if err != nil {
+				panic(err)
+			}
+		}
+		chains := p.parseIptablesChainObject(tableName, iptableBuffer.Bytes())
+		return iptable.NewIptables(tableName, chains)
+
+	}
+}
+
+// parseIptablesChainObject create a map of iptable chain name and iptable chain object
+func (p *Parser) parseIptablesChainObject(tableName string, byteArray []byte) map[string]*iptable.IptablesChain {
+	chainMap := make(map[string]*iptable.IptablesChain)
+	tablePrefix := []byte("*" + tableName)
+	curReadIndex := 0
+	for curReadIndex < len(byteArray) {
+		line, nextReadIndex := p.parseLine(curReadIndex, byteArray)
+		curReadIndex = nextReadIndex
+		if bytes.HasPrefix(line, tablePrefix) {
+			break
+		}
+	}
+
+	for curReadIndex < len(byteArray) {
+		line, nextReadIndex := p.parseLine(curReadIndex, byteArray)
+		curReadIndex = nextReadIndex
+		if len(line) == 0 {
+			continue
+		}
+		if bytes.HasPrefix(line, util.CommitBytes) || line[0] == '*' {
+			break
```

Comment:
if you break the for in this condition, you don't need the below else and can outdent that code